ext/net/ops_tls.rs

```diff
@@ -1075,8 +1082,25 @@ where
   let bind_addr = resolve_addr_sync(hostname, port)?
     .next()
     .ok_or_else(|| generic_error("No resolved address found"))?;
-  let std_listener = std::net::TcpListener::bind(bind_addr)?;
-  std_listener.set_nonblocking(true)?;
+  let domain = if bind_addr.is_ipv4() {
+    Domain::IPV4
+  } else {
+    Domain::IPV6
+  };
+  let socket = Socket::new(domain, Type::STREAM, None)?;
+  if reuse_address.is_some() {
+    socket.set_reuse_address(reuse_address.unwrap())?;
+  } else {
+    socket.set_reuse_address(true)?;
+  }
+  if reuse_port.is_some() {
+    socket.set_reuse_port(reuse_port.unwrap())?;
+  }
+  let socket_addr = socket2::SockAddr::from(bind_addr);
+  socket.bind(&socket_addr)?;
+  socket.listen(128)?;
+  socket.set_nonblocking(true)?;
```

Comment:
I believe these are only available for unix systems which causes CI failure

Reply:
I think this is the case. I noticed that **SO_REUSEADDR** in windows seems to achieve the same results of **SO_REUSEPORT** in Linux. I will add a `#[cfg(unix)]` to address this issue.
I also noticed that once swapped to `Socket2` for the creation of `TcpListener`, I had to set **SO_REUSEADDR** to `true` by default in order to pass the `net_test` unit tests.

Reply:
@Trolloldem that's a bit concerning. May I suggest to open a new PR that first switches to `socket2` crate and then add new APIs in follow up?

Reply:
That makes sense, that's why I'm suggesting to split the PR into two - so we can unsure that when changing the crates we keep the same behavior

Reply:
Ok, I will open a new PR with only the changes needed for `socket2`. When it is ready I will close this one, and propose the changes for **SO_REUSEADDR** in a new one

Reply:
Before starting to work on the new PR, I wanted to check how the current `std::net::TcpListener` is configured. I used the following code:

```
fn listen_tcp(
  state: &mut OpState,
  addr: SocketAddr,
) -> Result<(u32, SocketAddr), AnyError> {
  let std_listener1 = std::net::TcpListener::bind(&addr)?;
  let so: socket2::Socket = socket2::Socket::from(std_listener1);
  if so.reuse_address()? {
    panic!("The reuseADDR is: {}", so.reuse_address()?);
  }
  let std_listener: std::net::TcpListener = so.into();
  std_listener.set_nonblocking(true)?;
  let listener = TcpListener::from_std(std_listener)?;
  let local_addr = listener.local_addr()?;
  let listener_resource = TcpListenerResource {
    listener: AsyncRefCell::new(listener),
    cancel: Default::default(),
  };
  let rid = state.resource_table.add(listener_resource);

  Ok((rid, local_addr))
}
```
And the I run:
`target/debug/deno test --allow-all --unstable --location=http://js-unit-tests/foo/bar cli/tests/unit/net_test.ts `
The obtained output is the following:

```
running 33 tests from file:///home/oldem/git/deno/cli/tests/unit/net_test.ts
test netTcpListenClose ...
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.19.1
Args: ["target/debug/deno", "test", "--allow-all", "--unstable", "--location=http://js-unit-tests/foo/bar", "cli/tests/unit/net_test.ts"]

thread '<unnamed>' panicked at 'The reuseADDR is: true', ext/net/ops.rs:434:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

So, at least on Linux, it does seem that in this moment the socket already has SO_REUSEADDR option configured.
It is still preferable to separate the two PR?